cli/src/pcluster/config/cluster_config.py

```diff
@@ -65,49 +70,279 @@ def __repr__(self):
+        self.root_volume = root_volume
+        self.ephemeral_volume = ephemeral_volume
+
+
+class EfsConfig(Config):
+    """Represent the EFS configuration."""
+
+    def __init__(
+        self,
+        encrypted: bool = None,
+        kms_key_id: str = None,
+        performance_mode: str = None,
+        throughput_mode: str = None,
+        provisioned_throughput: int = None,
+        id: str = None,
+    ):
+        super().__init__()
+        if encrypted is None:
+            encrypted = MarkedBool(False)
+        if performance_mode is None:
+            performance_mode = MarkedStr("generalPurpose")
+        if throughput_mode is None:
+            throughput_mode = MarkedStr("bursting")
+        self.encrypted = encrypted
+        self.kms_key_id = kms_key_id
+        self.performance_mode = performance_mode
+        self.throughput_mode = throughput_mode
+        self.provisioned_throughput = provisioned_throughput
         self.id = id
-        self._validators = []
 
 
-class HeadNodeNetworkingConfig(Config):
-    """Represent the networking configuration for the Head Node."""
+class FsxConfig(Config):
+    """Represent the FSX configuration."""
+
+    def __init__(
+        self,
+        storage_capacity: str = None,
+        deployment_type: str = None,
+        export_path: str = None,
+        import_path: str = None,
+        imported_file_chunk_size: str = None,
+        weekly_maintenance_start_time: str = None,
+        automatic_backup_retention_days: str = None,
+        copy_tags_to_backups: bool = None,
+        daily_automatic_backup_start_time: str = None,
+        per_unit_storage_throughput: int = None,
+        backup_id: str = None,
+        kms_key_id: str = None,
+        file_system_id: str = None,
+        auto_import_policy: str = None,
+        drive_cache_type: str = None,
+        storage_type: str = None,
+    ):
+        super().__init__()
+        self.storage_capacity = storage_capacity
+        self.storage_type = storage_type
+        self.deployment_type = deployment_type
+        self.export_path = export_path
+        self.import_path = import_path
+        self.imported_file_chunk_size = imported_file_chunk_size
+        self.weekly_maintenance_start_time = weekly_maintenance_start_time
+        self.automatic_backup_retention_days = automatic_backup_retention_days
+        self.copy_tags_to_backups = copy_tags_to_backups
+        self.daily_automatic_backup_start_time = daily_automatic_backup_start_time
+        self.per_unit_storage_throughput = per_unit_storage_throughput
+        self.backup_id = backup_id
+        self.kms_key_id = kms_key_id
+        self.file_system_id = file_system_id
+        self.auto_import_policy = auto_import_policy
+        self.drive_cache_type = drive_cache_type
+        self.storage_type = storage_type
+        self._register_validator(FsxValidator, fsx_config=self)
+        # TODO register validators
```

Comment:
you can change this todo to "split validators"

Reply:
Done